### PR TITLE
chore!: rename shares to share

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: https://github.com/celestiaorg/go-square/issues/88

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ blob      | Package blob provides types and functions for working with blobs, bl
 inclusion | Package inclusion contains functions to generate the blob share commitment from a given blob.
 merkle    | Package merkle computes a deterministic minimal height Merkle tree hash.
 namespace | Package namespace contains the Namespace data structure.
-shares    | Package shares contains the Share data structure.
+shares    | package share contains the Share data structure.
```
Comment (rootulp) [inline README.md:13]:
```suggestion
share     | Package share contains the Share data structure.
```

File: share/blob.pb.go [hunk 2]
```diff
@@ -4,7 +4,7 @@
 // 	protoc        (unknown)
 // source: shares/blob.proto
```
Comment (rootulp) [inline share/blob.pb.go:5]: I think this prob needs a `make proto-gen` so that this comment gets updated

File: share/shares.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
-// Package shares contains the Share data structure.
-package shares
+// package share contains the Share data structure.
```
Comment (rootulp) [inline share/shares.go:1]: [optional][follow-up] we could expand this to:

```suggestion
// Package share contains the data structures: share, namespace, blob.
```

**Review by rootulp — APPROVED** [review]
> exported: type name will be used as share.ShareSequence by other packages, and that stutters; consider calling this Sequence (revive)

I agree with this suggestion but we def don't have to do it in this PR. IMO optional if we want to include in the next breaking release

File: README.md (hunk 1 above)
Comment (cmwaters) [reply README.md:13]: I should also remove the other packages that no longer exist. Thanks for pointing it out

**Comment by cmwaters** [comment]
> I agree with this suggestion but we def don't have to do it in this PR. IMO optional if we want to include in the next breaking release

I think golangci-lint will complain until this is fixed

**Comment by rootulp** [comment]
We could add a golangci-lint ignore directive if you want to merge this first and do a separate PR for the rename from `ShareSequence` -> `Sequence`

**Review by evan-forbes — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -6,11 +6,9 @@
 
 Package   | Description
 ----------|---------------------------------------------------------------------------------------------------------------------
-blob      | Package blob provides types and functions for working with blobs, blob transactions, and index wrapper transactions.
 inclusion | Package inclusion contains functions to generate the blob share commitment from a given blob.
 merkle    | Package merkle computes a deterministic minimal height Merkle tree hash.
-namespace | Package namespace contains the Namespace data structure.
-shares    | package share contains the Share data structure.
+share    | package share contains the Share data structure.
```
Comment (rootulp) [inline README.md:11]:
```suggestion
share    | Package share contains the Share data structure.
```

**Review by rootulp — APPROVED** [review]

**Review by rach-id — APPROVED** [review]

**Comment by cristaloleg** [comment]
Share is a very good name for a variable. Just saying. 

Reminds me net/url pain.

**Review by evan-forbes — APPROVED** [review]

**Comment by cmwaters** [comment]
> Share is a very good name for a variable. Just saying.
> 
> Reminds me net/url pain.

You're totally right. I had brought it up before when I wanted to call it something like `enc` for encoding or `split`

**Pull request merged**